File: backend/app/schemas/interview.py
from typing import Optional
from datetime import datetime
from enum import Enum
from pydantic import BaseModel, Field
from app.schemas.workflow import WorkflowConfig


class InterviewMessageRole(str, Enum):
    USER = "user"
    ASSISTANT = "assistant"


class InterviewMessage(BaseModel):
    id: Optional[str] = None
    role: InterviewMessageRole
    content: str
    created_at: datetime = Field(default_factory=datetime.utcnow)


class InterviewInput(BaseModel):
    user_message: str
    thinking_enabled: bool = False
    analysis_preferences: dict | None = None


class InterviewOutput(BaseModel):
    response: str
    is_complete: bool
    extracted_config: Optional[WorkflowConfig] = None
    suggested_competitors: list[str] = Field(default_factory=list)

File: frontend/types/interview.ts
import type { CompetitorGroups, WorkflowConfig } from "./workflow";

export type InterviewMessageRole = "user" | "assistant";

export interface InterviewMessage {
  id?: string;
  role: InterviewMessageRole;
  content: string;
  created_at: string;
}

export interface InterviewInput {
  user_message: string;
  thinking_enabled?: boolean;
  analysis_preferences?: {
    reportLanguage?: "中文" | "英文";
    analysisDepth?: "快速" | "标准" | "深入";
    competitorCount?: 3 | 5 | 8;
    outputFocus?: "功能" | "定价" | "用户反馈" | "全量";
    reportStyle?: "简报型" | "结构化研究型";
  };
}

export interface InterviewSSEMessage {
  token?: string;
  extracted_config?: Partial<WorkflowConfig>;
  suggested_competitors?: string[];
  suggested_competitor_groups?: CompetitorGroups;
  workflow_title?: string;
  is_complete?: boolean;
  response?: string;
}
